### DISCUSSION TITLE
How to explicitly mark an instance of frontend.Variable as constant, or create so?

### DISCUSSION THREAD

**Discussion by weijiguo** [Consensys/gnark#1041] [Q&A] [answered]
The data may have been processed somehow, hex-decoding, type conversion, etc. For example, we need to supply a `Domain Separation Tag` when calculating the `HashToG2` function. We need to make this `dst` a constant and ensure that its value impacts the circuit structure, rather than a witness which could be replaced with another value without impacting the circuit structure.

**Comment by gbotrel** [comment, marked as answer]
hi -- anything you pass to an frontend.API will be a constant (if it's not a variable :)). So:
```
api.Mul(1,2)
a := big.NewInt(...)
api.Mul(a, 2)
```
are all constant params.

**Reply by weijiguo** [reply]
Thanks, that's simple enough. I used to program with Arkworks and got used to creating a constant.
